### PR TITLE
TAS-2187/forget password

### PR DESCRIPTION
**Pull request by marjan-ara**

### CODE REVIEW

**Comment by satanmourner** [comment]
@marjan-ara could you please rebase/pull `main` branch into this PR?

File: src/core/adaptors/forgetPassword.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+export interface successRes {
+  message?: string;
+  error?: string;
+}
+
+export const forgetPasswordAdaptor = async (email: string): Promise<successRes> => {
```
Comment (satanmourner) [inline src/core/adaptors/forgetPassword.ts:6]: please use our `AdaptorRes` format as we discussed before

Comment (satanmourner) [reply src/core/adaptors/forgetPassword.ts:6]: line 20 too

File: src/modules/SignIn/EmptyPageLayout/index.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import BackLink from 'src/modules/General/components/BackLink';
+
+import css from './index.module.scss';
+import { EmptyPageLayoutProps } from './index.type';
+
+const EmptyPageLayout: React.FC<EmptyPageLayoutProps> = ({
```
Comment (satanmourner) [inline src/modules/SignIn/EmptyPageLayout/index.tsx:7]: export default please

File: src/pages/forgotPassword/newPassword/index.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import { useTranslation } from 'react-i18next';
+import Button from 'src/modules/General/components/Button';
+import FeaturedIcon from 'src/modules/General/components/FeaturedIcon';
+import Input from 'src/modules/General/components/Input';
+import EmptyPageLayout from 'src/modules/SignIn/EmptyPageLayout';
+
+import { useNewPassword } from './useNewPassword';
+
+export const NewPassword = () => {
+  const { t } = useTranslation();
```
Comment (satanmourner) [inline src/pages/forgotPassword/newPassword/index.tsx:11]: can you please use `translate` everywhere it's easier to search and find later

**Comment by satanmourner** [comment]
@marjan-ara please resolve conflicts too

File: .prettierrc.json [hunk 4]
```diff
@@ -2,14 +2,24 @@
...
   "printWidth": 120,
   "semi": true,
-  "tabWidth": 2,
+  "tabWidth": 4,
   "trailingComma": "all",
   "bracketSpacing": true,
   "bracketSameLine": false,
   "arrowParens": "avoid",
-  "endOfLine": "lf",
+  "endOfLine": "auto",
```
Comment (satanmourner) [inline .prettierrc.json:10]: why did you change prettier configs? it will be effected on everywhere

**Review by satanmourner — CHANGES_REQUESTED** [review]

File: .prettierrc.json (hunk 4 above)
Comment (marjan-ara) [reply .prettierrc.json:10]: this 'lf' causes problem for files on mac. 
But I will reverse it. I usually keep it on my machine.

File: src/pages/forgotPassword/email/index.tsx [hunk 5]
```diff
@@ -1,37 +1,36 @@
...
-          placeholder={translate('email-placeholder')}
-          errors={errors['email']?.message ? [errors['email']?.message.toString()] : undefined}
-        />
-        <Button variant="contained" color="primary" onClick={handleSubmit(handleForgetPassword)}>
-          {translate('password-reset')}
-        </Button>
-      </div>
-    </EmptyPageLayout>
-  );
+    const { t: translate } = useTranslation();
```
Comment (satanmourner) [inline src/pages/forgotPassword/email/index.tsx:10]: format of all your files in your last commit changed according to prettier configs, can you check all of them?

**Review by satanmourner — CHANGES_REQUESTED** [review]

**Review by satanmourner — APPROVED** [review]
